pwiz_tools/Skyline/Model/Proteome/ProteinAssociation.cs

```diff
@@ -223,6 +261,47 @@ private Dictionary<IProteinRecord, PeptideAssociationGroup> FindProteinMatches(I
             return proteinAssociations;
         }
 
+        private class ProteinPeptideMatches

```

Comment:
I'm thinking this class's interface could be a bit simpler, like enumerating a list of (string, bool) tuples (but probably with a defined type). Then the above code could be like:
for (var matchedPeptideInfo in proteinPeptideMatches)

Reply:
I changed it to use `Tuple<string, bool>` and eliminated about a dozen lines of code.